python/test/test_onnx.py

```diff
@@ -303,9 +304,9 @@ def onnx_convert(self, input_data: dict, graph_def, model_name: str):
         np.savez(input_npz, **input_data)
         # top mlir outputs will be inferenced first in case the quant mode is int8
         show_fake_cmd(input_npz, onnx_model, "onnx_out.npz")
-        onnx_outs = onnx_inference(input_data, onnx_model, True)
+        onnx_outs = onnx_inference(input_data, onnx_model, dump_all)
         show_fake_cmd(input_npz, fp32_mlir, "top_out.npz")
-        top_mlir_outs = mlir_inference(input_data, fp32_mlir, True)
+        top_mlir_outs = mlir_inference(input_data, fp32_mlir, dump_all)
```

Comment:
hi, it is an impressive solution for this task, but idk have you tried int8 quant mode test? the False dump_all will lead to missing calibration info of input, then make int8 lowering fail. So I think it would be better to modify the strategy a bit :)

Reply:
Yeah, I do failed int8 tests sometimes. I guess that a clever method of check_last is needed.

Reply:
a simpler possible way is you can change the locs so that comparison between certain ops will be skipped while you keep the dum_all as True. But we still need to make sure the final result is the same.

Reply:
I made a simple hack patch:

```
+        # this assumes that outputs are in order, i.e. the last one is the output
+        if check_last:
+            print("#######")
+            print(onnx_outs.keys())
+            print(top_mlir_outs.keys())
+            print("#######")
+            ok = list(onnx_outs.keys())[-1]
+            onnx_outs = {ok: onnx_outs[ok]}
+            tk = list(top_mlir_outs.keys())[-1]
+            top_mlir_outs = {ok: top_mlir_outs[tk]}
```

However, even with `dump_all=true`, I still met this:

```
tpuc-opt onnx_test_bm1684x/PermuteMove_Relu.mlir --init "--import-calibration-table=file=onnx_test_bm1684x/PermuteMove_Relu_cali_table asymmetric=True" --do-extra-converison=chip=bm1684x "--convert-top-to-tpu=mode=INT8 chip=bm1684x" --mlir-print-debuginfo --mlir-print-ir-after-all

init pass
....
do-extra-conversion pass

tpuc-opt: ../third_party/llvm/include/llvm/Support/Casting.h:567: decltype(auto) llvm::cast(const From&) [with To = mlir::quant::CalibratedQuantizedType; From = mlir::Type]: Assertion `isa<To>(Val) && "cast<Ty>() argument of incompatible type!"' failed.
```

I am not aware that if it is caused by qemu emulator.

Reply:
that problem is mainly caused by lacking calibration info of a certain op (inputOp mainly) in the calibration table (PermuteMove_Relu_cali_table in this case).  The quant::CalibratedQuantizedType requires min/ max/ threshold (see [ImportCalibrationTablePass](https://github.com/sophgo/tpu-mlir/blob/master/lib/Dialect/Top/Transforms/ImportCalibration.cpp) for more). 

You can manually add the calibration info in the cali table and try to run the cmd again to see if it works.
`tpuc-opt onnx_test_bm1684x/PermuteMove_Relu.mlir --init "--import-calibration-table=file=onnx_test_bm1684x/PermuteMove_Relu_cali_table asymmetric=True" --do-extra-converison=chip=bm1684x "--convert-top-to-tpu=mode=INT8 chip=bm1684x" --mlir-print-debuginfo --mlir-print-ir-after-all`

Reply:
It is fixed. PTAL again.